pkg/proxy/iptables/proxier.go

```diff
@@ -1337,6 +1338,19 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 		}
 
+		// Capture healthCheckNodePorts.
+		if svcInfo.HealthCheckNodePort() != 0 {
```

Comment:
should go inside the `if svcInfo.NodePort() != 0` section

Reply:
Node ports are now optional for load balancers (see `spec.allocateLoadBalancerNodePorts`), so a separate check still makes sense since healthCheckNodePorts is always set regardless of whether node ports is enabled.

Reply:
Second to @andrewsykim, I think these are effectively the same, for now, `HealthCheckNodePort` will be non-zero only if the svc is of `LoadBalancer` type. And if the service is of `LoadBalancer` type, then it should have a non-zero `NodePort`. To me it's api-server's job to validate the spec and ensure all the svc proxy sees are valid (see my previous [comment](https://github.com/kubernetes/kubernetes/pull/97824#issuecomment-756515241)). If couple the logic here then if someday the validation logic changed, then we also need to fix it here.